Products/ZenCallHome/CallHomeStatus.py

```diff
@@ -76,45 +78,45 @@ def load_from_redis(self, k="CallHomeStatus"):
             log.warning(
                 "Exception trying to recive data from redis: {0}".format(e))
             self._redis_client = None
-            return
+            return None
 
     def updateStat(self, param, value):
-        data = dict()
-        data = pickle.loads(self.load_from_redis())
+        data = self._pickleLoadRedis()
         data[param] = value
         self.push_to_redis(pickle.dumps(data))
 
     def getStat(self, param):
-        data = dict()
-        data = pickle.loads(self.load_from_redis())
+        data = self._pickleLoadRedis()
         return data.get(param)
 
     def getStatUI(self):
         """Returns status informations to UI
         """
         l = list()
-        data = dict()
-        data = pickle.loads(self.load_from_redis())
+        data = self._pickleLoadRedis()
         l.append({'id': 'lastsuccess', 'description': 'Last success', 'value': data.get('lastSuccess'), 'type': 'date'})
         l.append({'id': 'lastrun', 'description': 'Last run was', 'value': data.get('startedAt'), 'type': 'date'})
-        l.append({'id': 'lastupdtook', 'description': 'Last updating took', 'value': data.get('lastTook'), 'type': 'duration'})
+        l.append({'id': 'lastupdtook', 'description': 'Last updating took', 'value': data.get('lastTook'),
+                  'type': 'duration'})
         for key, val in data.iteritems():
             if isinstance(val, dict):
                 if val.get('status') == "FAILED":
-                    err = "Failed: "+val.get('error')
+                    err = "Failed: " + val.get('error')
                 else:
                     err = "No errors"
         l.append({'id': 'updresult', 'description': 'Updating result', 'value': err, 'type': 'text'})
         return l
 
+    def _pickleLoadRedis(self):
+        data = dict()
+        if self.load_from_redis() is not None:
+            data = pickle.loads(self.load_from_redis())
+        return data
```

Comment:
Please don't call `load_from_redis` twice.  Call it once and then check the result in a separate statement.